.github/workflows/preview.yml

```diff
@@ -14,51 +14,55 @@ jobs:
   Deploy-Preview:
     runs-on: ubuntu-latest
     steps:
+      - name: Find Comment
+        uses: peter-evans/find-comment@3eae4d37986fb5a8592848f6a574fdf654e61f9e # v3.1.0
+        id: fc
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          comment-author: 'github-actions[bot]'
+          body-includes: Preview Deployment
+
       - name: Comment PR
-        uses: thollander/actions-comment-pull-request@v2
+        uses: peter-evans/create-or-update-comment@71345be0265236311c031f5c7866368bd1eff043 # v4.0.0
+        id: comment
         with:
-          message: |
-            🌋🌋 Initializing deployment 🔃
-            🏗️🏗️ Building Next.js 🕧
-            🚀🚀 Deploying to Vercel 🕧
-            🛸🛸 Deployed to 🕧
-          comment_tag: preview_url
-
-      - name: Checkout main
-        uses: actions/checkout@v3
+          issue-number: ${{ github.event.pull_request.number }}
+          comment-id: ${{ steps.fc.outputs.comment-id }}
+          body: |
+            ## Preview Deployment
+            🌋 Initializing deployment 🔃
+            🏗️ Building Next.js 🕧
+            🚀 Deploying to Vercel 🕧
+            🛸 Deployed to 🕧
+          edit-mode: replace
+
+      - name: Checkout web repo
+        uses: actions/checkout@v4
         with:
             repository: ${{ secrets.WEB_REPO }}
             ref: main
             token: ${{ secrets.BOT_PAT }}
 
-      - name: Set pnpm
-        uses: pnpm/action-setup@v2
+      - name: Checkout self
+        uses: actions/checkout@v4
         with:
-          version: 9.x
+          path: ${{ github.event.repository.name }}
+
+      - name: Set pnpm
+        uses: pnpm/action-setup@a7487c7e89a18df4991f7f222e4898a00d66ddda # v4.1.0
 
       - name: Setup Node.js
-        uses: actions/setup-node@v3
+        uses: actions/setup-node@v4
         with:
           node-version: 20.x
-
-      - name: Cache node modules
-        id: cache-node-modules
-        uses: actions/cache@v3
-        with:
-          path: node_modules
-          key: node-modules-${{ hashFiles('./pnpm-lock.yaml') }}
+          cache: pnpm
 
       - name: Install dependencies
-        if: steps.cache-node-modules.outputs.cache-hit != true
         run: pnpm i --frozen-lockfile
 
       - name: Generate Design Tokens
         run: |
-          git clone --branch ${{ github.head_ref }} https://github.com/team-settle/settle-design-tokens.git
           pnpm design-tokens:process
-          rm -rf settle-design-tokens
-          git config user.name "GitHub Actions Bot"
-          git config user.email "<>"
           git checkout -b design-tokens-preview-${{ github.sha }}

```

Comment:
@carbonid1 I'm not sure this is actually used? I don't see a matching branch pattern on either repo, and I don't think we're pushing anywhere in this workflow?

Reply:
I vaguely remember this being a hack to make Vercel work. For some reason, I wanted to deploy this repo as if it were a settle-web repo. In any case, the end result is the deployment is assigned to https://vercel.com/settle/settle-web instead of a new project. The pro here is that the deployment will inherit all the env vars set up in that vercel project.

![CleanShot 2025-04-24 at 12 20 43](https://github.com/user-attachments/assets/f5d8d6ca-6cc9-48fa-8420-bedfcae7a83a)